index.js

```diff
@@ -123,63 +152,87 @@ module.exports = function (babel) {
         };
-      },
-
-      ImportDeclaration(path, state) {
-        let node = path.node;
 
+        // Setup other module options and create cache for values
         let modules = state.opts.modules || {
-          'htmlbars-inline-precompile': 'default',
+          'htmlbars-inline-precompile': { export: 'default', shouldParseScope: false },
         };
 
         if (state.opts.modulePaths) {
           let modulePaths = state.opts.modulePaths;
 
-          modulePaths.forEach((path) => (modules[path] = 'default'));
+          modulePaths.forEach((path) => (modules[path] = { export: 'default' }));
         }
 
-        let modulePaths = Object.keys(modules);
-        let matchingModulePath = modulePaths.find((value) => t.isLiteral(node.source, { value }));
-        let modulePathExport = modules[matchingModulePath];
+        let presentModules = new Map();
 
-        if (matchingModulePath) {
-          let first = node.specifiers && node.specifiers[0];
-          let localName = first.local.name;
+        for (let module in modules) {
+          let paths = importDeclarations.filter(
+            (path) => !path.removed && path.get('source').get('value').node === module
+          );
 
-          if (modulePathExport === 'default') {
-            if (!t.isImportDefaultSpecifier(first)) {
-              let input = state.file.code;
-              let usedImportStatement = input.slice(node.start, node.end);
-              let msg = `Only \`import hbs from '${matchingModulePath}'\` is supported. You used: \`${usedImportStatement}\``;
-              throw path.buildCodeFrameError(msg);
+          for (let path of paths) {
+            let { node } = path;
+            let options = modules[module];
+
+            // Normalize 'moduleName': 'importSpecifier'
+            if (typeof options === 'string') {
+              options = { export: options };
             }
-          } else {
-            if (!t.isImportSpecifier(first) || modulePathExport !== first.imported.name) {
-              let input = state.file.code;
-              let usedImportStatement = input.slice(node.start, node.end);
-              let msg = `Only \`import { ${modulePathExport} } from '${matchingModulePath}'\` is supported. You used: \`${usedImportStatement}\``;
 
-              throw path.buildCodeFrameError(msg);
+            let modulePathExport = options.export;
+
+            let first = node.specifiers && node.specifiers[0];
+            let localName = first.local.name;
+
+            if (modulePathExport === 'default') {
+              let importDefaultSpecifier = node.specifiers.find((n) =>
+                t.isImportDefaultSpecifier(n)
+              );
+
+              if (!importDefaultSpecifier) {
+                let input = state.file.code;
+                let usedImportStatement = input.slice(node.start, node.end);
+                let msg = `Only \`import ${first.local.name} from '${module}'\` is supported. You used: \`${usedImportStatement}\``;
```

Comment:
This seems vaguely odd, why do we reference `first` here? Can you update to either have the new modules format suggest a local name when the export is `default` (I think this option, but is pretty "legacy" IMO since we are generally going to use non-default exports going forward).